File: backend-java/src/main/java/org/egualpam/contexts/observable/shared/adapters/metrics/DomainEventMetrics.java
package org.egualpam.contexts.observable.shared.adapters.metrics;

import io.micrometer.core.instrument.Counter;
import io.micrometer.core.instrument.MeterRegistry;
import org.egualpam.contexts.observable.shared.application.domain.DomainEvent;

public class DomainEventMetrics {

    private final MeterRegistry meterRegistry;

    public DomainEventMetrics(MeterRegistry meterRegistry) {
        this.meterRegistry = meterRegistry;
    }

    public void published(DomainEvent event) {
        Counter.builder("application_domain_events_published")
                .tag("type", event.getClass().getSimpleName())
                .register(meterRegistry)
                .increment();
    }
}

File: backend-kotlin/src/main/kotlin/org/egualpam/contexts/observable/shared/adapters/metrics/DomainEventMetrics.kt
package org.egualpam.contexts.observable.shared.adapters.metrics

import io.micrometer.core.instrument.Counter
import io.micrometer.core.instrument.MeterRegistry
import org.egualpam.contexts.observable.shared.application.domain.DomainEvent

class DomainEventMetrics(
  private val meterRegistry: MeterRegistry
) {
  fun published(domainEvent: DomainEvent) {
    Counter.builder("application_domain_events_published")
        .tag("type", domainEvent.javaClass.simpleName)
        .tag("app", "order-service-backend-kotlin")
        .register(meterRegistry)
        .increment()
  }
}
